socs/agents/site_camera/agent.py

```diff
@@ -0,0 +1,216 @@
+import txaio
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+
+# For logging
+txaio.use_twisted()
+
+
+class SiteCameraAgent:
+    """Grab screenshots from ACTi site cameras.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    camera_addresses : list
+        List of IP addresses (as strings) of cameras.
+    user : str
+        Username of cameras.
+    password : str
+        Password of cameras.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    is_streaming : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the UPS. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+    """
+
+    def __init__(self, agent, camera_addresses, user, password):
+        self.agent = agent
+        self.is_streaming = False
+        self.log = self.agent.log
+        self.lock = TimeoutLock()
+
+        self.cameras = []
+        for camera in camera_addresses:
+            self.cameras.append({'address': camera,
+                                 'connected': True})
+        self.user = user
+        self.password = password
+
+        self.lastGet = 0
+
+        agg_params = {
+            'frame_length': 10 * 60  # [sec]
+        }
+        self.agent.register_feed('cameras',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    def acq(self, session, params=None):
+        """acq()
+
+        **Process** - Fetch values from the UPS via SNMP.
+
+        Parameters
+        ----------
+        test_mode : bool, optional
+            Run the Process loop only once. Meant only for testing.
+            Default is False.
+
+        Notes
+        -----
+        The most recent data collected is stored in session.data in the
+        structure::
+
+            >>> response.session['data']
+            # for each camera
+            {'camera1_last_attempt': 1701983575.032506,
+             'camera1_connected': True,
+             'camera1_address': '10.10.10.41'}
+        """
+
+        session.set_status('running')
+        self.is_streaming = True
+        count = 0
+        while self.is_streaming:
+            current_time = time.time()
+            # Check if 60 seconds has passed before getting screenshot
+            if (current_time - self.lastGet) < 60:
+                continue
+
+            # Use UTC
+            timestamp = datetime.now(timezone.utc).replace(tzinfo=timezone.utc).timestamp()
+            data = {
+                'block_name': 'cameras',
+                'timestamp': timestamp,
+                'data': {}
+            }
+
+            count += 1
+            for i, camera in enumerate(self.cameras):
```

Comment:
I think the most static naming would be location (container1, satp1, etc...). This could be another argument list that goes along with the IP addresses?